src/LinAlg/hiopMatrixDenseRowMajor.hpp

```diff
@@ -208,12 +208,12 @@ class hiopMatrixDenseRowMajor : public hiopMatrixDense
   virtual long long get_local_size_m() const { return m_local_; }
   virtual MPI_Comm get_mpi_comm() const { return comm_; }
 
-  //TODO: this is not kosher!
-  inline double** local_data() const {return M_; }
-  inline double*  local_buffer() const {return M_[0]; }
+  double* local_data_const() const {return M_[0]; }
```

Comment:
Perhaps there is no need to create an array to row pointers anymore? How about just creating a contiguous data block for the matrix data?